packages/ui/src/coreModules/form/higherOrderComponents/injectSearchOptions.js

```diff
@@ -214,9 +214,11 @@ const injectSearchOptions = (
       response = [],
       { skipPlainTextOption = false, value } = {}
     ) {
-      if (Array.isArray(response)) {
-        const { searchQuery } = this.state
-
+      const { searchQuery } = this.state
+      if (
+        (searchQuery && Array.isArray(response)) ||
+        (!searchQuery && response.length === 1)
```

Comment:
is this intended to show an already selected result? in that case maybe it's more correct to do `(!searchQuery && value)`?